added tdid and ad-unit-code

jrosendahl opened the pull request:
<!--
Thank you for your pull request. Please make sure this PR is scoped to one change, and that any added or changed code includes tests with greater than 80% code coverage. See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [ x ] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: if checking here, also submit your bidder params documentation here https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
<!-- Describe the change proposed in this pull request -->

<!-- For new bidder adapters, please provide the following -->
- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

## Other information
<!-- References to related PR or issue #s, @mentions of the person or team responsible for reviewing changes, etc. -->

Commit:
added tdid and ad-unit-code

Commit:
fixed tdid

Commit:
merged master

pm-harshad-mane committed:
PubMatic bid adapter to support price floors module (#5387)

* added support for pubcommon, digitrust, id5id

* added support for IdentityLink

* changed the source for id5

* added unit test cases

* changed source param for identityLink

* added support for floors module

* using floorModule to set floor

* removed commented console.log statements

colbertk committed:
Triplelift Adaptor: Use Floors Module (#5329)

* access floor from floors module

* int -> float and null check

* let -> const

Commit:
removed digitrust

jsnellbaker committed:
appnexusBidAdapter - fix video params (#5394)

* appnexusBidAdapter - fix video params

* remove mimes field

pm-harshad-mane committed:
PubMatic adds support for bidUserIdAsEids (#5397)

* added support for pubcommon, digitrust, id5id

* added support for IdentityLink

* changed the source for id5

* added unit test cases

* changed source param for identityLink

* id value should be string only

* id value should be string; changed test data

* pubmatic using userIdAsEids

commented unnecessary code
PubMatic will no longer read Digitrust from config or independent library setup
PubMatic will no loner read TTDID from config or independent library setup

* chnages in test cases for userIdAsEids

commented unnecessary test cases
PubMatic will no longer read Digitrust from config or independent library setup
PubMatic will no loner read TTDID from config or independent library setup

* deleted commented code and test cases

* deleted a commented line

* lint effect

jrosendahl committed:
repush

weswhitney commented on test/spec/modules/sovrnBidAdapter_spec.js:
```diff
@@ -70,12 +70,17 @@ describe('sovrnBidAdapter', function() {
       expect(payload.imp[0].banner.h).to.equal(1)
     })
 
+    it('includes the ad unit code int the request', function() {
```
typo

weswhitney reviewed (approved):
saw a typo on test suite description.

lowendavid committed:
External ids (#5351)

* externalIds (SIM-657) Added external ids and associated unit test.

* externalIds - Added DEFAULT_PARAMS_WITH_EIDS that contains the external id information for the unit test

* externalIds (SIM-657) uses createEidsArray found in the eid.js

* externalIds cleanup spaces and tabs, adding a missing import for createEids in eids.js

* externalIds adding a space where it is needed to fix a wonderful test.

curlyblueeagle committed:
33Across: CCPA Compliance + Schain support (#5365)

* check gdpr in buildRequest

* User sync based on whether gdpr applies or not

* check if consent data exists during user sync

* split user sync into further branches: 1) when gdpr does not apply 2) when consent data is unavailable

* contribute viewability to ttxRequest

* update tests

* remove window mock from tests

* use local variables

* introduce ServerRequestBuilder

* add withOptions() method to ServerRequestBuilder

* add semicolons

* sync up package-lock.json with upstream/master

* stub window.top in tests

* introduce getTopWindowSize() for test purpose

* reformat code

* add withSite() method to TtxRequestBuilder

add withSite() method to TtxRequestBuilder

* add isIframe() and _isViewabilityMeasurable()

* handle NON_MEASURABLE viewability in nested iframes

* consider page visibility, stub utils functions getWindowTop() and getWindowSelf()

* contribute viewability as 0 for inactive tab

* add prebidjs version to ttx request

* send caller as an array

* fix JSDoc in utils.js

* send viewability as non measurable when unable to locate target HTMLElement, add warning message

* introduce mapAdSlotPathToElementId()

* introduce getAdSlotHTMLElement(), add logging

* introduce mapAdSlotPathToElementId()

* update logging in ad unit path to element id mapping

* rephrase logging, fix tests

* update adapter documentation

* remove excessive logging

* improve logging

* revert change

* fix return of _mapAdUnitPathToElementId()

* improve logging of _mapAdUnitPathToElementId()

* do not use Array.find()

* return id once element is found

* return id once element is found

* let -> const

* Removing killswitch behavior for GDPR

* Updated comments to reflect current gdpr logic

* URI encode consent string

* Updated example site ID to help Prebid team e2e test our adapter

* send page url in ortb

* Removed redundant pageUrl default

* Restored package-log.json that mirrors prebid's repo

* Sending USP string during buildRequest

* Adding USP consent data to user sync

* add unit test for syncing without bidrequest

* Changed to uspConsent to make the connatation consistent

* Resetting adapter state in adapter after user sync rather than exposing it.

* removed console log

* Adding schain info

* remove setting empty format ext

* better tests invalid values

* removing validation of schain

* Fixed lint errors

Co-authored-by: Gleb Glushtsov <gleb.glushtsov@33across.com>
Co-authored-by: Gleb Glushtsov <glebglushtsov@users.noreply.github.com>
Co-authored-by: Gleb Glushtsov <gleb.glushtsov@gmail.com>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-29-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@admins-MacBook-Pro.local>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-137-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-96-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@AHEGDE-MAC.local>
Co-authored-by: Aparna Hegde <ahegde@AHEGDE-MAC.fios-router.home>

mshuhaliia committed:
added waardex adapter (#5349)

* added waardex adapter

* removed support http from adapter

Co-authored-by: Max Shuhaliia <mshuhaliia@Maxs-MacBook-Pro-2.local>

mirkorean committed:
Add customParams to yieldlab configuration (#5374)

Customers are using custom parameters to our endpoint to use them downstream. We need a way of supporting this in prebid.

uditalias committed:
Vidazoo Adapter: Feature/user-id (#5386)

* feat(module): multi size request

* fix getUserSyncs
added tests

* update(module): package-lock.json from master

* feat(client): add user id request support

* fix(client): lint errors

Co-authored-by: roman <shmoop207@gmail.com>

jankoulaga committed:
LiveIntentId submodule updates (#5407)

* LiveIntentId submodule.

Bumped the live-connect-js version with changes:
- fixed the problem where userId/index.js was sending the full config on `decode`, while the docs claim it should only send `.params`
- removed the uuid lib
- removed the support for legacy LI first party cookies
- removed the redundant config parameter `providedIdentifierName`
- support for running live-connect inside of multiple wrappers

* LiveIntentId submodule.

Removed reading of non-accessible and therefore redundant code snippets

* Removed excessive config logging.

* Making the publisher id parameter optional.

* Removed unneeded check.

curlyblueeagle committed:
33Across: Adding floors support (#5408)

* check gdpr in buildRequest

* User sync based on whether gdpr applies or not

* check if consent data exists during user sync

* split user sync into further branches: 1) when gdpr does not apply 2) when consent data is unavailable

* contribute viewability to ttxRequest

* update tests

* remove window mock from tests

* use local variables

* introduce ServerRequestBuilder

* add withOptions() method to ServerRequestBuilder

* add semicolons

* sync up package-lock.json with upstream/master

* stub window.top in tests

* introduce getTopWindowSize() for test purpose

* reformat code

* add withSite() method to TtxRequestBuilder

add withSite() method to TtxRequestBuilder

* add isIframe() and _isViewabilityMeasurable()

* handle NON_MEASURABLE viewability in nested iframes

* consider page visibility, stub utils functions getWindowTop() and getWindowSelf()

* contribute viewability as 0 for inactive tab

* add prebidjs version to ttx request

* send caller as an array

* fix JSDoc in utils.js

* send viewability as non measurable when unable to locate target HTMLElement, add warning message

* introduce mapAdSlotPathToElementId()

* introduce getAdSlotHTMLElement(), add logging

* introduce mapAdSlotPathToElementId()

* update logging in ad unit path to element id mapping

* rephrase logging, fix tests

* update adapter documentation

* remove excessive logging

* improve logging

* revert change

* fix return of _mapAdUnitPathToElementId()

* improve logging of _mapAdUnitPathToElementId()

* do not use Array.find()

* return id once element is found

* return id once element is found

* let -> const

* Removing killswitch behavior for GDPR

* Updated comments to reflect current gdpr logic

* URI encode consent string

* Updated example site ID to help Prebid team e2e test our adapter

* send page url in ortb

* Removed redundant pageUrl default

* Restored package-log.json that mirrors prebid's repo

* Sending USP string during buildRequest

* Adding USP consent data to user sync

* add unit test for syncing without bidrequest

* Changed to uspConsent to make the connatation consistent

* Resetting adapter state in adapter after user sync rather than exposing it.

* removed console log

* Adding schain info

* remove setting empty format ext

* better tests invalid values

* removing validation of schain

* Fixed lint errors

* First cut for bidfloors support

* fixed where getFloors is read

* fixed merge conflicts

Co-authored-by: Gleb Glushtsov <gleb.glushtsov@33across.com>
Co-authored-by: Gleb Glushtsov <glebglushtsov@users.noreply.github.com>
Co-authored-by: Gleb Glushtsov <gleb.glushtsov@gmail.com>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-29-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@admins-MacBook-Pro.local>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-137-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-96-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@AHEGDE-MAC.local>
Co-authored-by: Aparna Hegde <ahegde@AHEGDE-MAC.fios-router.home>

uditalias committed:
Vidazoo Adapter: Feature/screen-size (#5385)

* feat(module): multi size request

* fix getUserSyncs
added tests

* update(module): package-lock.json from master

* feat(client): add screen resolution as request param

* feat(client): add `res` param to spec

* fix(client): screen size inside test

Co-authored-by: roman <shmoop207@gmail.com>

justinkuo-appier committed:
Appier: add support for aliases (#5392)

* Add initial partial implementation for Appier bidder adapter.

* Use relative protocol for bidder API url.

* Handle server response for Appier adapter and add related unit tests.

* Support farm-specific prebid server and allow overriding the server with setConfig().

* Add doc for Appier bid adapter.

* Fix const correctness.

* Append requestId to the beacon image URL of Appier adapter to reduce the risks of being cached by proxy servers or browsers.

* Send bidderRequest.refererInfo to Appier bidder server.

* Remove the show beacon since now we generate it in the backend server.

* Only generate a show beacon url if it's not provided by the backend.

* Add version information for Appier adapter using semver (starts from 1.0.0).

* Add a new adapter for Appier bidder.

* Add a new adapter for Appier bidder.

* add appier analyticsAdapter skeleton.

* update initial working version.

* refactor and remove debug messages.

* fix config checking logic.

* implement bidAdjustment and add timeout before send event.

* unify cache operation into cacheManager to avoid direct key/value operation.

* Update server name.

* correct currency impl and message payload. remove unused debug messages.

* update var naming.

* rename creative vars. cleanup comments.

* add test cases for AnalyticsAdapter.

* update test specs file.

* remove spec from main branch.

* add unit tests for AnalyticsAdapter, #1 - Happy cases.

* refactor tests.

* update unit tests for AnalyticsAdapter.

* add tests for bid-adjusted, bid-timeout events

* fix bid adjustment test case.

* add nobid case in analytics unit test.

* add test case for delayed bids and prebidWon messages.

* Use logInfo and logError utilility functions instead of console.log() to print debug messages.

* handle timeout status message correctly.

* correct isTimeout setting logic and test cases.

* replace for...of by array.forEach for IE11

* apply auto formatter.

* refactor: extract timeout logic to a helper function

* tag analytic version with 0.1.0-beta for iCook release

* Re-implement the appier analytics adapter with a simpler design and correctly handle timeouts.

* Remove unused variables.

* Rename methods to improve consistency.

* Code cleanup: rename methods and avoid duplicated code.

* Fix wrong bid response data caused by non-deterministic event ordering of prebid.js.

* Send bid message immediately on auction end without delay.

* add withCredentials in ajax call to get client cookie.

* Fix broken unit test for appier analytic adapter.

* add prediction id support.

* update predictionId/configId format in tests.

* Init refined appier analytics testing

* hotfix: appier analytics support browserstack bug

* feat: add aliases

Co-authored-by: Hong Jen-Yee (PCMan) <pcman.hong@appier.com>
Co-authored-by: Yuan-Hung Huang <yuanhung.huang@appier.com>
Co-authored-by: kdchang <kd.chang@appier.com>
Co-authored-by: chih-ping-weng <chihping.weng@appier.com>

arasaki-yuki committed:
Add GMOSSP Adapter (#5377)

* Add GMOSSP Adapter

* fix test param

jrosendahl committed:
add package-lock from upstream  master

jrosendahl committed:
Delete package-lock.json

jrosendahl committed:
add package-lock from upstream  master

jrosendahl committed:
Merge remote-tracking branch 'upstream/master' into SNEK-320-TDID
